examples/dodge_the_creeps/src/hud.rs

```diff
@@ -0,0 +1,88 @@
+use crate::extensions::NodeExt as _;
+use gdnative::*;
+
+#[derive(NativeClass)]
+#[inherit(CanvasLayer)]
+#[user_data(user_data::ArcData<HUD>)]
+#[register_with(Self::register_hud)]
+pub struct HUD;
+
+#[methods]
+impl HUD {
+    fn register_hud(builder: &init::ClassBuilder<Self>) {
+        builder.add_signal(init::Signal {
+            name: "start_game",
+            args: &[],
+        });
+    }
+
+    fn _init(_owner: CanvasLayer) -> Self {
+        HUD
+    }
+
+    #[export]
+    pub unsafe fn show_message(&self, owner: CanvasLayer, text: String) {
+        // let mut message_label = owner
+        //     .get_node("message_label".into())
+        //     .expect("Missing message_label")
+        //     .cast::<Label>()
+        //     .expect("Cannot cast to Label");
+
+        let mut message_label: Label = owner
+            .get_typed_node("message_label")
+            .expect("Cannot cast to Label");
+
+        message_label.set_text(text.into());
+        message_label.show();
+
+        owner
+            .get_typed_node::<Timer, _>("message_timer")
+            .expect("Cannot cast to Timer")
+            .start(0.0);
+    }
+
+    pub unsafe fn show_game_over(&self, owner: CanvasLayer) {
+        self.show_message(owner, "Game Over".into());
+        // yield($message_timer, "timeout")
```

Comment:
I just realized that I did not deal with any of the `yield`s. I am having trouble finding the equivalent in this crate. Would anyone be able to point out what I should be doing here?

Reply:
Unfortunately, there is no way to conveniently wait for a signal in Rust, currently. Support using the recently stabilized async-await feature is planned for 0.9, as detailed in #284, but not implemented yet.

I think the example runs reasonably well even without the yields, so for now you can just remove them.